### ISSUE TITLE
Fetch works in browser but not in desktop application

### ISSUE THREAD

**Issue by JanTrichter** [OfficeDev/office-js#2755]
Hello there,
I was recently creating an add-in for Outlook and I switched to the web for debugging purposes. It works great now and I managed to fix all errors except when trying to run the application in the desktop client.

I get `failed to fetch` as an error message. The API that I call has a custom SSL certificate which caused problems in the browser earlier but is not causing any problems currently (in the browser at least).

## Desktop Environment (not working)
* Platform: PC Desktop
* Host: Outlook
* Office version number: Microsoft 365 MSO (Version 2206 Build 16.0.15330.20260) 64 Bit (Desktop)
* Operating System: Windows 10 build19044

## Browser Environment (working)
* Platform: Web
* Host: Outlook
* Office version number: --
* Operating System: Windows 10 build19044
* Browser: Google Chrome 103.0.5060.134
 
## Expected behavior
Fetch should return endpoint's data.

## Current behavior
Error: `failed to fetch`

## Steps to reproduce
1. Create outlook extension that uses `fetch()`
2. Sideload add in in the browser and desktop
3. try if `fetch()` works in both

## Link to live example
Can't provide one unfortunately :(

## Useful logs
Only error: `failed to fetch`

**Comment by timwan10** [comment]
Are you trying to do this in a ShowTaskpane or ExecuteFunction based add-in? or a LaunchEvent Add-in?

**Comment by JanTrichter** [comment]
@timwan10 `<Action xsi:type="ShowTaskpane">` if that's what you mean.

**Comment by timwan10** [comment]
Does adding the appropriate domains to AppDomains work 
* https://docs.microsoft.com/en-us/javascript/api/manifest/appdomains?view=excel-js-preview
* https://docs.microsoft.com/en-us/javascript/api/manifest/appdomain?view=excel-js-preview

**Comment by JanTrichter** [comment]
@timwan10 already did that but it has no effect.

**Comment by timwan10** [comment]
At this point you may need to run fiddler and capture your queries, and see if that leads to any additional information as to why it's failing.

**Comment by JanTrichter** [comment]
@timwan10 I managed to make it work. First I had to make sure that both the host for the static files was using http (Outlook denied localhost for whatever reason) and then make sure I added the API as AppDomain (already did that before). Lastly I had to create a valid ssl certificate for the API so that it wouldn't fail to fetch. Looks like Outlook cached an old version or I forgot to upload the new version to the remote server which caused the `fail to fetch` due to invalid certificates in the first place.

Debugging it with MS Edge allowed me to find the error more easily.

Thanks for your help!
